### DISCUSSION TITLE
Download/Export Loaded Audio

### DISCUSSION THREAD

**Discussion by r-cha** [katspaugh/wavesurfer.js#3343] [Q&A]
Hi,

WaveSurfer handles fetching audio from a url, which is great 👍

But I want to provide a "Download" button for my users, and avoid the redundant download from the original URL. I followed along with #2972 (and a few other similar questions) and tried to make it non-region specific, but it's not quite what I want due to having to process the buffer into a wav. I want the exact original downloaded content.

I just assume the blob url that WaveSurfer uses is available somewhere, and I want to set it as the href for a button. Is that possible?

Thanks

**Comment by katspaugh** [comment]
You can pass a custom media element and access the src of that. There’s no way to get the blob directly.

**Reply by r-cha** [reply]
Thank you @katspaugh! I ended up fetching directly and then using `load(blobURL)` for WaveSurfer.
